## Task

Fixni: app-1  | Error: Cannot find module '@prisma/engines'
app-1  | Require stack:
app-1  | - /app/node_modules/prisma/build/index.js
app-1  |     at Function._resolveFilename (node:internal/modules/cjs/loader:1383:15)
app-1  |     at defaultResolveImpl (node:internal/modules/cjs/loader:1025:19)
app-1  |     at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1030:22)
app-1  |     at Function._load (node:internal/modules/cjs/loader:1192:37)
app-1  |     at TracingChannel.traceSync (node:diagnostics_channel:328:14)
app-1  |     at wrapModuleLoad (node:internal/modules/cjs/loader:237:24)
app-1  |     at Module.require (node:internal/modules/cjs/loader:1463:12)
app-1  |     at require (node:internal/modules/helpers:147:16)
app-1  |     at Object.<anonymous> (/app/node_modules/prisma/build/index.js:239:4130)
app-1  |     at Module._compile (node:internal/modules/cjs/loader:1706:14) {
app-1  |   code: 'MODULE_NOT_FOUND',
app-1  |   requireStack: [ '/app/node_modules/prisma/build/index.js' ]
app-1  | }
app-1  | 
app-1  | Node.js v22.22.0
app-1  | Running database migrations...
app-1  | node:internal/modules/cjs/loader:1386
app-1  |   throw err;
app-1  |   ^
app-1  | 
app-1  | Error: Cannot find module '@prisma/engines'
app-1  | Require stack:
app-1  | - /app/node_modules/prisma/build/index.js
app-1  |     at Function._resolveFilename (node:internal/modules/cjs/loader:1383:15)
app-1  |     at defaultResolveImpl (node:internal/modules/cjs/loader:1025:19)
app-1  |     at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1030:22)
app-1  |     at Function._load (node:internal/modules/cjs/loader:1192:37)
app-1  |     at TracingChannel.traceSync (node:diagnostics_channel:328:14)
app-1  |     at wrapModuleLoad (node:internal/modules/cjs/loader:237:24)
app-1  |     at Module.require (node:internal/modules/cjs/loader:1463:12)
app-1  |     at require (node:internal/modules/helpers:147:16)
app-1  |     at Object.<anonymous> (/app/node_modules/prisma/build/index.js:239:4130)
app-1  |     at Module._compile (node:internal/modules/cjs/loader:1706:14) {
app-1  |   code: 'MODULE_NOT_FOUND',
app-1  |   requireStack: [ '/app/node_modules/prisma/build/index.js' ]
app-1  | } ... vyzkousej pak sam, ze to jde uspesne spustit v dockeru a ze to nehaze chyby criticke

--- a/Dockerfile
+++ b/Dockerfile
@@ -2,6 +2,10 @@
 FROM node:22-alpine AS deps
 WORKDIR /app
 
+# OpenSSL must be present so npm postinstall for @prisma/engines downloads
+# the linux-musl-openssl-3.0.x binaries (matching the runner environment).
+RUN apk add --no-cache openssl
+
 COPY package.json package-lock.json ./
 RUN npm ci
 
@@ -38,16 +42,19 @@ COPY --from=builder /app/.next/standalone ./
 COPY --from=builder /app/.next/static ./.next/static
 COPY --from=builder /app/public ./public
 
-# Copy Prisma schema (needed for migrate deploy) and generated engines
+# Copy Prisma schema (needed for migrate deploy) and generated client
 COPY --from=builder /app/prisma ./prisma
+# Generated client with musl engine binary (produced by prisma generate in builder)
 COPY --from=builder /app/node_modules/.prisma ./node_modules/.prisma
 
-# Copy all @prisma/* packages (client + engines + debug + get-platform + etc.)
-COPY --from=builder /app/node_modules/@prisma ./node_modules/@prisma
+# Copy all @prisma/* packages from the deps stage (pristine npm ci on Alpine+OpenSSL).
+# Using deps (not builder) guarantees @prisma/engines contains the correct
+# linux-musl-openssl-3.0.x binaries without any post-build mutations.
+COPY --from=deps /app/node_modules/@prisma ./node_modules/@prisma
 
-# Prisma CLI is needed to run migrate deploy at startup
-COPY --from=builder /app/node_modules/prisma ./node_modules/prisma
-COPY --from=builder /app/node_modules/.bin/prisma ./node_modules/.bin/prisma
+# Prisma CLI is needed to run migrate deploy at startup — copy from deps too
+COPY --from=deps /app/node_modules/prisma ./node_modules/prisma
+COPY --from=deps /app/node_modules/.bin/prisma ./node_modules/.bin/prisma
 
 # SQLite database lives here; mount a volume over this path in production
 RUN mkdir -p /app/data && chown -R nextjs:nodejs /app/data /app/node_modules

